File: src/kyu_5/josephus_permutation/csharp/josephus_permutation.cs
using System;
using System.Collections.Generic;
public class Josephus
{
   public static List<object> JosephusPermutation(List<object> items, int k)
   {
       List<object> res = new List<object>();
       int pos = 0;
       while (items.Count > 0)
       {
           pos = (pos + k - 1) % items.Count;
           res.Add(items[pos]);
           items.RemoveAt(pos);
       }
       return res;
   }
}

File: src/kyu_5/josephus_permutation/cpp/josephus_permutation.cpp
#include <vector>

std::vector <int> josephus(std::vector <int> items, int k) {
    std::vector< int > result;
    int pos = k-1;

    while (items.size())
    {
        pos %= items.size();
        result.push_back(items[pos]);
        items.erase(items.begin()+pos);
        pos += k-1;
    }

    return result;
}